File: March_2022/24_03_2022/Search_Titles_of_Books.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <string.h>

using namespace std;

int main(){
    string a;
    int n;

    getline(cin,a);
    cin>>n;

    for(int i=0;i<a.length();i++){
        a[i] = tolower(a[i]);
    }

    char data[n][100];
    char ans[n][100];

    for(int i=0;i<n;i++){
        cin.getline(data[i],100);
    }

    int masterIndex = 0;

    for(int i=0;i<n;i++){
        bool flag = true;
        for(int j=0;j<strlen(data[i]);j++){
            char x = tolower(data[i][j]);
            if(a.find(x) == string::npos){
                flag = false;
                break;
            }
        }
        if(flag){
            strcpy(ans[masterIndex++],data[i]);
        }
        cout<<data[i]<<endl;
    }

    if(masterIndex != 0){
        for(int i=0;i<masterIndex;i++){
            cout<<ans[i]<<endl;
        }
    }
    else{
        cout<<"No such book"<<endl;
    }

    return 0;
}

File: March_2022/24_03_2022/Search_Titles_of_Books.py
a = input().lower()
n = int(input())
data = [input() for i in range(n)]
interest = []
for i in data:
    flag = True
    for j in i:
        x = j.lower()
        if(x not in a):
            flag = False
            break
    if(flag):
        interest.append(i)
if(len(interest) != 0):
    for i in interest:
        print(i)
else:
    print("No such book")